### PR TITLE
✨  Support using system truststore to verify the hub certificate

### PR DESCRIPTION
**Pull request by zhujian7**
issue reference: https://issues.redhat.com/browse/ACM-13373

### CODE REVIEW

File: pkg/bootstrap/bootstrapkubeconfig.go [hunk 1]
```diff
@@ -281,15 +360,32 @@ func getKubeAPIServerCAData(ctx context.Context, clientHolder *helpers.ClientHol
...
 	}
-	return []byte(rootCA.Data["ca.crt"]), nil
+
+	if len(caKeys) == 0 {
+		caKeys = []string{"ca-bundle.crt", "ca.crt", "tls.crt"}
+	}
+
+	for _, key := range caKeys {
+		if data, ok := rootCA.Data[key]; ok {
+			return []byte(data), nil
```
Comment (elgnay) [inline pkg/bootstrap/bootstrapkubeconfig.go:384]: If multiple keys exit, the values should be merged.

Comment (zhujian7) [reply pkg/bootstrap/bootstrapkubeconfig.go:372]: ok, but why users put multiple bundles into different keys？ in that case, should they use multiple configmaps?

Comment (elgnay) [reply pkg/bootstrap/bootstrapkubeconfig.go:372]: Yes, you are right. It's might not be a good idea to merge the value of different keys.

File: pkg/bootstrap/bootstrapkubeconfig.go [hunk 2]
```diff
@@ -230,14 +236,75 @@ func GetBootstrapCAData(ctx context.Context, clientHolder *helpers.ClientHolder,
 
 func getKubeAPIServerCAData(ctx context.Context, clientHolder *helpers.ClientHolder, kubeAPIServer string,
 	caNamespace string, klusterletConfig *klusterletconfigv1alpha1.KlusterletConfig) ([]byte, error) {
+
+	if klusterletConfig != nil && klusterletConfig.Spec.HubKubeAPIServerConfig != nil {
+		return getKubeAPIServerCADataFromConfig(ctx, clientHolder, kubeAPIServer, caNamespace,
+			klusterletConfig.Spec.HubKubeAPIServerConfig)
+	}
+
+	// TODO: DEPRECATE the following code and only use the HubKubeAPIServerConfig in the future
```
Comment (xuezhaojun) [inline pkg/bootstrap/bootstrapkubeconfig.go:245]: Can we add `@zhujian7` on this todo for quicker understanding in the future?

File: pkg/bootstrap/bootstrapkubeconfig.go [hunk 3]
```diff
@@ -402,6 +486,14 @@ func GetProxySettings(klusterletConfig *klusterletconfigv1alpha1.KlusterletConfi
...
 		return "", nil
 	}
+
+	if klusterletConfig.Spec.HubKubeAPIServerConfig != nil &&
+		len(klusterletConfig.Spec.HubKubeAPIServerConfig.ProxyURL) > 0 {
+		// the TrustedCABundles configured in the HubKubeAPIServerConfig is already added by the getKubeAPIServerCAData
+		return klusterletConfig.Spec.HubKubeAPIServerConfig.ProxyURL, nil
+	}
+
+	// TODO: DEPRECATE the following code and only return the proxyURL
```
Comment (xuezhaojun) [inline pkg/bootstrap/bootstrapkubeconfig.go:496]: Same as above.

**Comment by zhujian7** [comment]
Hi @xuezhaojun , the prow will auto squash the commits.

**Pull request merged**